Flaky AdvertisedAddressTest shouldCommunicateOverProxy

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? First time
- Does it block your work? No
- Do we suspect that it is a real failure?

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
2022-08-02T09:08:04.7805530Z [ERROR] io.camunda.zeebe.it.network.AdvertisedAddressTest.shouldCommunicateOverProxy  Time elapsed: 66.902 s  <<< FAILURE!
2022-08-02T09:08:04.7806671Z java.lang.AssertionError: Expected all partitions to be healthy, but partition <1> of broker <1> is <UNHEALTHY>
2022-08-02T09:08:04.7807632Z 	at io.camunda.zeebe.it.network.AdvertisedAddressTest.shouldCommunicateOverProxy(AdvertisedAddressTest.java:110)
2022-08-02T09:08:04.7808619Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-08-02T09:08:04.7809512Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-08-02T09:08:04.7813437Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-08-02T09:08:04.7814413Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-08-02T09:08:04.7815299Z 	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
2022-08-02T09:08:04.7816276Z 	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
2022-08-02T09:08:04.7817566Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
2022-08-02T09:08:04.7819212Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
2022-08-02T09:08:04.7820177Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
2022-08-02T09:08:04.7821308Z 	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
2022-08-02T09:08:04.7822188Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
2022-08-02T09:08:04.7823199Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
2022-08-02T09:08:04.7824214Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
2022-08-02T09:08:04.7827552Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
2022-08-02T09:08:04.7829142Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
2022-08-02T09:08:04.7830455Z 	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
2022-08-02T09:08:04.7832316Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
2022-08-02T09:08:04.7833582Z 	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
2022-08-02T09:08:04.7834924Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
2022-08-02T09:08:04.7836171Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7837706Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
2022-08-02T09:08:04.7839119Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
2022-08-02T09:08:04.7840409Z 	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
2022-08-02T09:08:04.7841845Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
2022-08-02T09:08:04.7843524Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7844843Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-08-02T09:08:04.7846309Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-08-02T09:08:04.7847465Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-08-02T09:08:04.7848827Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7850188Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-08-02T09:08:04.7851571Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-08-02T09:08:04.7853203Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
2022-08-02T09:08:04.7855178Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
2022-08-02T09:08:04.7856813Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-08-02T09:08:04.7858176Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7859402Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-08-02T09:08:04.7860504Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-08-02T09:08:04.7861813Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-08-02T09:08:04.7863095Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7864363Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-08-02T09:08:04.7865608Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-08-02T09:08:04.7867250Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
2022-08-02T09:08:04.7869364Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
2022-08-02T09:08:04.7871181Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2022-08-02T09:08:04.7872466Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7873716Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2022-08-02T09:08:04.7874762Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2022-08-02T09:08:04.7875924Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2022-08-02T09:08:04.7877169Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2022-08-02T09:08:04.7878477Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2022-08-02T09:08:04.7879691Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2022-08-02T09:08:04.7881567Z 	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
2022-08-02T09:08:04.7883373Z 	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
2022-08-02T09:08:04.7884323Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
2022-08-02T09:08:04.7885193Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
2022-08-02T09:08:04.7886340Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
2022-08-02T09:08:04.7887148Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
2022-08-02T09:08:04.7888137Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
</pre>
</details>

**Logs**
https://github.com/camunda/zeebe/runs/7628085698?check_suite_focus=true

--- a/qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/AdvertisedAddressTest.java
+++ b/qa/integration-tests/src/test/java/io/camunda/zeebe/it/network/AdvertisedAddressTest.java
@@ -107,7 +107,9 @@ final class AdvertisedAddressTest {
               .map(ContainerProxy::getOriginalProxyPort)
               .toList();
       TopologyAssert.assertThat(topology)
-          .isComplete(3, 1, 3)
+          .hasClusterSize(3)
+          .hasExpectedReplicasCount(1, 3)
+          .hasLeaderForEachPartition(1)
           .hasBrokerSatisfying(
               b ->
                   assertThat(b.getAddress())

